src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -1133,16 +1133,19 @@ FORCEINLINE bool Thread::InlineTryFastReversePInvoke(ReversePInvokeFrame * pFram
     // a do not trigger mode.  The exception to the rule allows us to have [UnmanagedCallersOnly] methods that are called via
     // the "restricted GC callouts" as well as from native, which is necessary because the methods are CCW vtable
     // methods on interfaces passed to native.
-    if (IsCurrentThreadInCooperativeMode())
+    // We will allow threads in DoNotTriggerGc mode to do reverse PInvoke regardless of their coop state.
+    if (IsDoNotTriggerGcSet())
```

Comment:
I think we want to do this before the `TSF_Attached` check. We do not want to attach untached GC threads as a side-effect.

Reply:
Right, on server GC this could be unattached thread.

Reply:
is there a way to pass `ServerGarbageCollection=true` to `src\tests\build.cmd` ?

Reply:
You can set `DOTNET_gcServer=1`

Reply:
Turns out `src\tests\build.cmd` can pass through paramters to msbuild, but `src/tests/build.proj` was not propagating some parameters that could be useful.